src/components/frame/Extensions/pages/MultiSigPage/hooks.ts

```diff
@@ -0,0 +1,161 @@
+import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { ExtensionSaveSettingsContext } from '~context/ExtensionSaveSettingsContext/ExtensionSaveSettingsContext.ts';
+import { ActionTypes } from '~redux/actionTypes.ts';
+import { isInstalledExtensionData } from '~utils/extensions.ts';
+
+import {
+  MultiSigThresholdType,
+  type DomainThresholdConfig,
+  type MultiSigSetupFormValues,
+} from './types.ts';
+import {
+  getDomainThresholds,
+  getInitialDomainConfig,
+  getInitialGlobalThreshold,
+  getThresholdType,
+} from './utils.ts';
+
+export const useBackNavigation = ({ extensionData }) => {
+  const navigate = useNavigate();
+
+  /*
+   * If we arrive here but the extension is not installed go back to main extension details page
+   */
+  useEffect(() => {
+    if (!isInstalledExtensionData(extensionData)) {
+      navigate(-1);
+    }
+  }, [extensionData, navigate]);
+};
+
+export const useThresholdData = ({ extensionData }) => {
+  const multiSigConfig = extensionData.params?.multiSig || null;
+
+  const {
+    colony: { colonyAddress, domains },
+  } = useColonyContext();
+
+  const {
+    register,
+    setValue,
+    getValues,
+    trigger,
+    formState: { errors, isValid },
+    reset,
+  } = useForm<MultiSigSetupFormValues>({
+    defaultValues: {
+      globalThreshold: getInitialGlobalThreshold(multiSigConfig),
+    },
+    mode: 'onChange',
+  });
+
+  const { handleIsVisible, handleSetActionType, resetAll, callback } =
+    useContext(ExtensionSaveSettingsContext);
+
+  const [thresholdType, setThresholdType] = useState<MultiSigThresholdType>(
+    getThresholdType(multiSigConfig?.colonyThreshold),
+  );
+
+  const [domainThresholdConfigs, setDomainThresholdConfigs] = useState<
+    DomainThresholdConfig[]
+  >([]);
+
+  useEffect(() => {
+    handleIsVisible(true);
+    handleSetActionType(ActionTypes.MULTISIG_SET_THRESHOLDS);
+
+    return () => {
+      resetAll();
+      reset();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    setValue('colonyAddress', colonyAddress);
+  }, [colonyAddress, setValue]);
+
+  useEffect(() => {
+    if (!domains || !domains.items || !multiSigConfig) {
+      return;
+    }
+
+    const domainsExcludingRoot = domains.items.filter(
+      (domain): domain is NonNullable<typeof domain> =>
+        domain !== null && !domain.isRoot,
+    );
+
+    const tempDomainThresholdConfigs = domainsExcludingRoot.map((domain) => {
+      const { threshold, ...rest } = getInitialDomainConfig(
+        domain,
+        multiSigConfig,
+      );
+
+      setValue(rest.name, threshold);
+
+      return rest;
+    });
+
+    setDomainThresholdConfigs(tempDomainThresholdConfigs);
+  }, [domains, multiSigConfig, setValue]);
+
+  useImperativeHandle(callback, () => ({
```

Comment:
I think it would be better to refer to `callback` as `callbackRef` here, just for a bit of clarity since this is after all supposed to be a ref

And because we're already exploring the depths of refs here, if you have time, I think it would be great to give this hook a type so we know exactly what the name of the exposed functions are 😄